File: Python/Linked List/Linear Linked List/Reversing a Linked List.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

# Creation of Linked List starts
class LinkedList:
    def __init__(self,list1):
        self.head = None
        self.head=Node(list1[0])
    
    def Append(self,list1):
      last=Node(None)
      last=self.head
      
      for i in range(1,len(list1),1):
       t         = Node(list1[i])
       last.next = t
       last      = t
# Creation of Linked List stops
   
    def Reverse(self):
     p = self.head
     q = None
     r = None
     while p:
         r=q
         q=p
         p=p.next
         q.next=r
     self.head=q 
     
    def Display(self):
         ptr=self.head
         while ptr:
             print(ptr.data,end=" ")
             ptr = ptr.next

list1=[23,34,45,67,100]
obj1=LinkedList(list1)
obj1.Append(list1)

print("Before Reversing : ",end="")
obj1.Display()

obj1.Reverse()

print()
print("After Reversing  : ",end="")
obj1.Display()

File: C++/Practise/Reversing a linked list.c
#include<stdio.h>
#include<stdlib.h>

struct Node
 {
 	int data;
 	struct Node *next; 	
 }*first=NULL;
 
 void Create(int A[],int size)
  {
  	struct Node*last,*t;
  	first=(struct Node*)malloc(sizeof(struct Node));
  	first->data=A[0];
  	first->next=NULL;
  	last=first;
	 
	 int i;
	 for(i=1;i<size;i++)
	   {
	   	struct Node *t=(struct Node*)malloc(sizeof(struct Node));
	   	t->data=A[i];
	   	t->next=NULL;
	   	last->next=t;
	   	last=t;
	   }
}

void Display(struct Node *ptr)
 {
 	while(ptr!=NULL)
 	 {
 	 	printf("%d",ptr->data);
 	 	ptr=ptr->next;
	 } 	
 }
 
void Reverse(struct Node *p)
{
	struct Node *q=NULL;
	struct Node *r=NULL;
	
	while(p!=NULL)
	  {
	  	r=q;
	  	q=p;
	  	p=p->next;
	  	q->next=r;	  	
	  }
	  first=q;
}
	   	
int main()
 {
 	int n,i;
 	printf("Enter the size of the array : ");
 	scanf("%d",&n);
 	int A[n];
 	printf("Enter the array elements : ");
 	for(i=0;i<n;i++)
 	 {
 	 	scanf("%d",&A[i]);
	 }
 	
 	Create(A,n);
 	printf("\nBefore Reversing : ");
	Display(first);
 	Reverse(first);
 	printf("\nAfter Reversing : ");
	Display(first);
 }
